syncer/syncer_test.go

```diff
@@ -983,3 +1000,227 @@ func (s *testSyncerSuite) TestGeneratedColumn(c *C) {
 	}
 	s.catchUpBinlog()
 }
+
+func (s *testSyncerSuite) TestcheckpointID(c *C) {
+	syncer := NewSyncer(s.cfg)
+	checkpointID := syncer.checkpointID()
+	c.Assert(checkpointID, Equals, "101")
+}
+
+func (s *testSyncerSuite) TestExecErrors(c *C) {
+	syncer := NewSyncer(s.cfg)
+	syncer.appendExecErrors(new(ExecErrorContext))
+	c.Assert(syncer.execErrors.errors, HasLen, 1)
+
+	syncer.resetExecErrors()
+	c.Assert(syncer.execErrors.errors, HasLen, 0)
+}
+
+func (s *testSyncerSuite) TestCasuality(c *C) {
+	s.cfg.WorkerCount = 1
+	syncer := NewSyncer(s.cfg)
+	syncer.jobs = []chan *job{make(chan *job, 1)}
+
+	go func() {
+		job := <-syncer.jobs[0]
+		c.Assert(job.tp, Equals, flush)
+		syncer.jobWg.Done()
```

Comment:
any synchronization	mechanism to ensure this goroutine executed?

Reply:
what do you mean?

Reply:
I mean, this goroutine may not execute if the main goroutine exited, then we do not whether `c.Assert(job.tp, Equals, flush)` passed.

Reply:
you are right, done